packages/react-devtools-shared/src/backend/views/TraceUpdates/index.js

```diff
@@ -86,6 +96,11 @@ function traceUpdates(nodes: Set<HostInstance>): void {
       rect = measureNode(node);
     }
 
+    let displayName = agent.getComponentNameForHostInstance(node);
```

Comment:
Good point. Do you think there’s a cleaner way to check if the component was optimized by the compiler? I saw the `parseElementDisplayNameFromBackend` utility can return `compiledWithForget`, but it needs the`type`. Not sure if I can get that from `HostNode` since it’s typed as an object.

Reply:
> Not sure if I can get that from HostNode since it’s typed as an object.

Here we operate with `HostInstance`: `HTMLElement` for Web (with `react-dom`), and something completely different for Native, based on version and architecture of React Native. 

You could leave it like this for now, maybe later we could add a new method to `Agent`, which would map `HostInstance` to `DevToolsInstance` (this is what `getComponentNameForHostInstance` is already doing), and then return if component is compiled or not, and maybe filter out HOC names.